### PR TITLE
8258732: runtime/cds/appcds/dynamicArchive/DynamicArchiveRelocationTest.java fails

### PR DESCRIPTION
**Pull request by calvinccheung**
On Windows, sometimes the OS picks 0x0000000800000000 as the requested archive base address. This resulting in runtime relocation won't happen. The test expects the `Unmapping region #3` output trace which won't be there if there no relocation. The fix is to adjust the test to look for another output trace in case the matching of the original trace has failed.

Testing: tier 1 - 4.
Also ran the test 50 times with tier4 vm options on Windows.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258732](https://bugs.openjdk.java.net/browse/JDK-8258732): runtime/cds/appcds/dynamicArchive/DynamicArchiveRelocationTest.java fails

### Reviewers
 * [Yumin Qi](https://openjdk.java.net/census#minqi) (@yminqi - **Reviewer**) ⚠️ Review applies to dd802cad8c9d85f6e746d4592a2999bf0a876dab
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2415/head:pull/2415`
`$ git checkout pull/2415`

### CODE REVIEW

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/DynamicArchiveRelocationTest.java [hunk 1]
```diff
@@ -91,8 +91,13 @@ static void doTest() throws Exception {
         String topArchiveName  = getNewArchiveName("top");
 
         String runtimeMsg = "Try to map archive(s) at an alternative address";
+        String runtimeRelocMsg = "runtime archive relocation start";
         String unmapPrefix = ".*Unmapping region #3 at base 0x.*";
         String unmapPattern = unmapPrefix + "(Bitmap)";
+        String archiveRelocPattern = ".*ArchiveRelocationMode == 1.*";
+        String unmapRgn1Pattern = ".*Unmapping region #1 at base 0x.*";
+        String unmapRgn0Prefix = ".*Unmapping region #0 at base 0x.*";
```
Comment (yminqi) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/DynamicArchiveRelocationTest.java:99]: unmapRgn0Prefix is only used to form unmapRgn0Pattern, they could merge together.

**Review by yminqi — APPROVED** [review]
LGTM.

**Review by iklam — APPROVED** [review]

**Pull request closed without merging**